lib/services/publish-service.ts

```diff
@@ -0,0 +1,208 @@
+                private $projectFilesManager: Project.IProjectFilesManager,
+                private $project: Project.IProject,
+                private $prompter: IPrompter,
+                private $server: Server.IServer) {
+
+        this.$project.ensureProject();
+
+        if (!this.$project.capabilities.publish) {
+            this.$errors.failWithoutHelp("The publish command is only applicable to projects of type %s", this.$projectConstants.TARGET_FRAMEWORK_IDENTIFIERS.MobileWebsite);
+        }
+
+        this.readConnections().wait();
+        this.ftpPublishConnections = _.filter(this.allPublishConnections, c => c.type === PublishService.ALLOWED_CONNECTION_TYPE);
+    }
+
+    public publish(idOrUrl: string, username: string, password: string): IFuture<void> {
+        return (() => {
+            var projectName = this.$project.projectData.ProjectName;
+            var ftpConnectionData: Server.OfflineClientFtpConnectionData = {
+                RemoteUrl: idOrUrl,
+                ShouldPurge: options.force,
+                Username: username,
+                Password: password
+            };
+
+            if (!ftpConnectionData.Username) {
+                ftpConnectionData.Username = this.$prompter.getString("Username:").wait();
+            }
+
+            if (!ftpConnectionData.Password) {
+                ftpConnectionData.Password = this.$prompter.getPassword("Password:", {allowEmpty: true}).wait();
+            }
+
+            var id = parseInt(idOrUrl) - 1;
+
+            // the check whether it contains a dot is performed
+            // in case the user tries to publish to an IP address
+            if (!isNaN(id) && idOrUrl.indexOf('.') < 0) {
+                this.validateId(id);
+
+                ftpConnectionData.RemoteUrl = this.ftpPublishConnections[id].publishUrl;
+            }
+
+            var projectDir = this.$project.getProjectDir().wait();
+            this.$projectFilesManager.excludeFile(projectDir, PublishService.JSON_PUBLISH_FILE_NAME);
+            this.$project.importProject().wait();
+            this.$logger.printInfoMessageOnSameLine("Publishing");
+
+            try {
+                this.$progressIndicator.showProgressIndicator(this.$server.publish.publishToFtp(projectName, projectName, ftpConnectionData), 6000).wait();
+            } catch (err) {
+                this.$logger.printInfoMessageOnSameLine(os.EOL);
+                if (err.message === PublishService.ERROR_MESSAGE_WHEN_PROJECT_EXISTS_ON_REMOTE) {
+                    this.$errors.failWithoutHelp("This project already exists on the remote server. " + os.EOL + 
+                                    "To remove all files and upload the new ones on the remote run $ appbuilder publish remove --force");
+                }
+
+                throw err;
+            }
+
+            this.$logger.printInfoMessageOnSameLine(os.EOL);
+            this.$logger.info("Project '%s' published successfully.", projectName);
```

Comment:
Is it published or uploaded? I am almost certain that uploaded is the more correct term here.